### PR TITLE
aws may transiently refuse to recognize a new pending instance

### PR DESCRIPTION
**Pull request by avg-I**
We are making two calls right after the instance creation,
`boto.ec2.instance.update()` and `boto.ec2.connection.create_tags()`,
so these could be affected by the problem.

Now, if we get `InvalidInstanceID.NotFound` error we simply retry the
affected calls.

https://clusterhq.atlassian.net/browse/FLOC-3537

### CODE REVIEW

File: flocker/provision/_aws.py [hunk 1]
```diff
@@ -67,7 +79,13 @@ def _wait_until_running(instance):
...
             ):
                 sleep(1)
-            instance.update()
+            try:
+                instance.update()
+            except EC2ResponseError as e:
+                _check_response_error(
+                    e,
+                    u"flocker:provision:aws:wait_until_running:retry"
+                )
```
Comment (exarkun) [inline flocker/provision/_aws.py]: Likewise - `flocker.common.poll_until` with a timeout might be a good thing to use here.

**Comment by exarkun** [comment]
Thanks.  Please give the retrying helpers a try and then merge.

File: flocker/provision/_aws.py (hunk 1 above)
Comment (avg-I) [inline flocker/provision/_aws.py]: JP, you said likewise, but I am not sure likewise to what :-)
In any case, I''ll try `flocker.common.poll_until`

Comment (exarkun) [inline flocker/provision/_aws.py]: Uh. :/  Looks like github lost my other comment.  I'll try to re-create it.

File: flocker/provision/_aws.py [hunk 2]
```diff
@@ -242,10 +260,17 @@ def create_node(self, name, distribution,
...
+            while True:
+                try:
+                    self._connection.create_tags([instance.id], metadata)
+                    break
+                except EC2ResponseError as e:
+                    _check_response_error(
+                        e,
+                        u"flocker:provision:aws:create_node:retry"
+                    )
+                    sleep(1)
```
Comment (exarkun) [inline flocker/provision/_aws.py]: I guess it was about using `poll_until` or a similar helper here.

**Comment by avg-I** [comment]
@exarkun I've implemented your suggestion about `poll_until`. The code changed in a slightly non-trivial way, so could you please take another look just in case?
Thanks!

**Pull request merged**